openapi_python_client/parser/openapi.py

```diff
@@ -258,37 +279,21 @@ def _add_parameters(
             if isinstance(prop, ParseError):
                 return ParseError(detail=f"cannot parse parameter of endpoint {endpoint.name}", data=prop.data), schemas
 
-            if prop.python_name in used_python_names:
-                duplicate, duplicate_location = used_python_names[prop.python_name]
-                if duplicate.python_name == prop.python_name:  # Existing should be converted too for consistency
-                    duplicate.set_python_name(f"{duplicate.python_name}_{duplicate_location}", config=config)
-                prop.set_python_name(f"{prop.python_name}_{param.param_in}", config=config)
-            else:
-                used_python_names[prop.python_name] = (prop, param.param_in)
+            for location, parameters_dict in parameters_by_location.items():
+                existing_prop: Optional[Property] = parameters_dict.get(prop.name)
+                same_location: bool = location == param.param_in
 
-            endpoint.relative_imports.update(prop.get_imports(prefix="..."))
+                if existing_prop and not same_location:
```

Comment:
I think there's a potential for naming conflicts causing issues when the location _is_ the same. For example, if you had "some_name" in both path and query, it would be renamed to "some_name_path" and "some_name_query". If you then had a param named "some_name_path" in path, I think you'd get invalid Python code.

This should either error out here or attempt to further resolve naming issues. I think the former is prudent since we could spend all day trying to de-conflict things and it probably wouldn't be consistent.

Perhaps this was the idea behind `used_python_identifiers`? So new names can be checked?

Reply:
Oops! Sure, I really forgot to add the `used_python_identifiers` check. Sorry. It is added now. Also I added a test for your case.

To resolve conflicts, I wanted to always use the `{param.name}_{param.param_in}` for the parameter identifiers. But I think this is a rare use case. So I put it back as it was before. 

I think it's a good idea to make it an option in the config. Something like `parameters_location_postfix: bool`. This will allow avoiding the use of postfixes with non-duplicated identifiers in most cases. And this will allow avoiding naming conflicts in the case you described, if this option is set.

I added the `parameters with same Python identifier` error. Maybe we should automatically retry generate endpoint with option described above if that error happens.